solutions/cf-1536C.mdx

```diff
@@ -0,0 +1,151 @@
+---
+id: cf-1536C
+source: CF
+title: Diluc and Kaeya
+author: Kevin Sheng
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/91520)
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+#include <unordered_map>
+#include <algorithm>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+int main() {
+	int test_num;
+	std::cin >> test_num;
+	for (int t = 0; t < test_num; t++) {
+		int len;
+		std::cin >> len;
+		vector<char> str(len);
+		for (char& c : str) {
+			std::cin >> c;
+		}
+		
+		int d_num = 0;
+		int k_num = 0;
+		vector<int> max_pref_chunks;
+		std::unordered_map<int, std::unordered_map<int, int>> prev_ratios;
+		for (char c : str) {
+			if (c == 'D') {
+				d_num++;
+			} else if (c == 'K') {
+				k_num++;
+			}
+			// get the simplified ratio by dividing both quantities by the gcd
+			int gcd = std::__gcd(d_num, k_num);
+			int d_ratio = d_num / gcd;
+			int k_ratio = k_num / gcd;
+			// add the ratio to the records and record the max prefix chunk amount
+			max_pref_chunks.push_back(++prev_ratios[d_ratio][k_ratio]);
```

Comment:
why not just `cout  << ++prev_ratios[d_ratio][k_ratio]` here

Reply:
whitespace is an absolute prick